src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8479,7 +8484,7 @@ protected MethodGroupResolution ResolveExtension(
             OverloadResolution.Options options,
             RefKind returnRefKind,
             TypeSymbol? returnType,
-            ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo,
```

Comment:
> ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo [](http://example.com/codeflow?start=11&length=52)

Consider keeping this parameter and collecting lookup use-site info it.

Reply:
As discussed as follow-up for previous PR, it's not clear why some use-site info should be reported in this new channel that we introduced vs. added to the diagnostics reported as part of the returned `MethodGroupResolution`. What's the split/rationale you're suggesting?
The model offered here follows the design from `main` (we collect use-site diagnostics and report them as part of the returned result).

Reply:
I think that it makes sense to always report use-site errors that potentially caused lookup to fail. It is quite possible that we would want to do the same for the process of figuring out applicability of candidates, but I am fine with separating this two